src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -175,7 +175,8 @@ int MacroAssembler::patch_oop(address insn_addr, address o) {
   // instruction.
   if (Instruction_aarch64::extract(insn, 31, 21) == 0b11010010101) {
     // Move narrow OOP
-    narrowOop n = CompressedOops::encode((oop)o);
+    narrowOop noop = CompressedOops::encode((oop)o);
+    uint32_t n = CompressedOops::narrow_oop_value(noop);
```

Comment:
I wonder if we should simply create an overload for oop so that this (and the other places) could be changed to:
```uint32_t n = CompressedOops::narrow_oop_value((oop)o);```